### PR TITLE
Initial code

### PR DESCRIPTION
**Pull request by bibendi**
https://jira.railsc.ru/browse/PC4-18137

### CODE REVIEW

File: spec/lib/sysloggable/logger_spec.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+  it "counts duration" do
+    Timecop.freeze(Time.now)
+
+    expect(syslogger).to receive(:add).
+      with(described_class::SEVERITIES[:info],
+           "severity=INFO service=test_ident operation= duration=10.0 message=msg")
+
+    logger.info("msg") do
+      Timecop.freeze(Time.now + 10)
```
Comment (apress-hound) [inline spec/lib/sysloggable/logger_spec.rb]: Do not use `Time.now` without zone. Use one of `Time.zone.now`, `Time.now.current`, `Time.now.in_time_zone`, `Time.now.utc`, `Time.now.getlocal`, `Time.now.iso8601`, `Time.now.jisx0301`, `Time.now.rfc3339`, `Time.now.to_i`, `Time.now.to_f` instead.

File: spec/lib/sysloggable/logger_spec.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+      expect(syslogger).to receive(:add).
+        with(severity_value, "severity=#{sev_label} service=test_ident operation= duration=0 message=msg")
+
+      logger.send(severity_name, "msg")
+    end
+  end
+
+  it "counts duration" do
+    Timecop.freeze(Time.now)
```
Comment (apress-hound) [inline spec/lib/sysloggable/logger_spec.rb]: Do not use `Time.now` without zone. Use one of `Time.zone.now`, `Time.now.current`, `Time.now.in_time_zone`, `Time.now.utc`, `Time.now.getlocal`, `Time.now.iso8601`, `Time.now.jisx0301`, `Time.now.rfc3339`, `Time.now.to_i`, `Time.now.to_f` instead.

File: lib/sysloggable/logger.rb [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      @logger.push_tags(@options[:tags]) if @options[:tags]
+
+      @logger
+    end
+
+    def add(severity, message, params = {})
+      if block_given?
+        beginning = Time.now
+        yield params
+        duration = Time.now - beginning
```
Comment (apress-hound) [inline lib/sysloggable/logger.rb]: Do not use `Time.now` without zone. Use one of `Time.zone.now`, `Time.now.current`, `Time.now.in_time_zone`, `Time.now.utc`, `Time.now.getlocal`, `Time.now.iso8601`, `Time.now.jisx0301`, `Time.now.rfc3339`, `Time.now.to_i`, `Time.now.to_f` instead.

File: lib/sysloggable.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+require "sysloggable/version"
+require "sysloggable/container"
+
+module Sysloggable
+  def self.InjectLogger(options)
```
Comment (apress-hound) [inline lib/sysloggable.rb:5]: Use snake_case for method names.

Comment (bibendi) [reply lib/sysloggable.rb:5]: да да да, я знаю, так нада

File: lib/sysloggable/logger.rb [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+      warn: ::Logger::WARN,
+      error: ::Logger::ERROR,
+      unknown: ::Logger::UNKNOWN
+    }.freeze
+
+    def initialize(options)
+      @options = options
+    end
+
+    [:debug, :info, :warn, :error, :fatal, :unknown].each do |logger_method|
```
Comment (artofhuman) [inline lib/sysloggable/logger.rb]: fatal нет в `SEVERITIES` че то не понял как это будет работать.

Comment (bibendi) [reply lib/sysloggable/logger.rb]: fixed

Comment (artofhuman) [reply lib/sysloggable/logger.rb]: тут может тогда лучше `SEVERITIES.keys` ?

File: lib/sysloggable/logger.rb [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+        severity: ::Logger::SEV_LABEL[severity],
+        service: @options.fetch(:service_name, @options.fetch(:ident)),
+        operation: params.delete(:operation),
+        duration: duration,
+        message: message
+      }.merge!(params)
+
+      result.each_with_object(String.new) do |(key, value), memo|
+        memo << " " << "#{key}=#{value}"
+      end.strip
```
Comment (artofhuman) [inline lib/sysloggable/logger.rb]: тут вроже можно `strip!`

Comment (bibendi) [reply lib/sysloggable/logger.rb]: fixed

File: lib/sysloggable/logger.rb (hunk 5 above)
Comment (bibendi) [reply lib/sysloggable/logger.rb]: fixed

**Review by artofhuman — APPROVED** [review]

**Comment by bibendi** [comment]
next /cc @abak-press/sysloggable-owners

File: lib/sysloggable.rb (hunk 4 above)
Comment (Napolskih) [reply lib/sysloggable.rb:5]: нахватался фишек :smile_cat:

**Review by Napolskih — APPROVED** [review]

**Pull request merged**